server/src/main/java/org/elasticsearch/common/blobstore/BlobContainer.java

```diff
@@ -111,9 +112,12 @@
 
     /**
      * Deletes this container and all its contents from the repository.
+     * @param resultConsumer Long consumer that implementations can invoke with the byte size of each deleted blob when deleting it.
```

Comment:
I'm not super happy about having to do it in this fuzzy way here, but I don't see a way of doing it any other way on HDFS without a massive performance hit (HDFS has a recursive delete RPC call that we run one-off but that doesn't return any statistics about what was deleted itself). IMO, this is fine though, for users running this it should mostly be relevant whether or not anything was cleaned up so they can tweak the frequency of running this endpoint accordingly (if they don't experience any failures during snapshotting or deleting snapshots they should never need this endpoint and always getting `0` returns from it will transport that message just fine I hope).